hrms/setup.py

```diff
@@ -586,7 +594,14 @@ def create_user_type(user_type, data):
 
 
 def create_role_permissions_for_doctype(doc, data):
-	for doctype, perms in data.get("doctypes").items():
+	append_user_doctypes_to_user_type(data.get("doctypes").items(), doc)
+
+	if doc.role == "Employee Self Service" and "lending" in frappe.get_installed_apps():
+		append_user_doctypes_to_user_type(get_lending_doctypes_for_ess().items(), doc)
+
+
+def append_user_doctypes_to_user_type(doctypes, doc):
+	for doctype, perms in doctypes:
```

Comment:
```suggestion
	append_docperms_to_user_type(data.get("doctypes"), doc)

	if doc.role == "Employee Self Service" and "lending" in frappe.get_installed_apps():
		append_docperms_to_user_type(get_lending_doctypes_for_ess(), doc)


def append_docperms_to_user_type(data: dict, doc: dict):
	for doctype, perms in data.items():
```

- always try and keep minimum preprocessing for the calling functions. The function being called should handle all common actions. Eg: the `.items()` operation here
- Changed the name from `append_user_doctypes_to_user_type` to `append_docperms_to_user_type` to signify what it exactly does